rust/lance/src/index.rs

```diff
@@ -604,18 +623,49 @@ impl DatasetIndexExt for Dataset {
                 }
                 Ok(sum)
             })
-            .collect::<Result<Vec<_>>>()?;
+            .collect::<Result<Vec<_>>>();
+
+        async fn migrate_and_recompute(ds: &Dataset, index_name: &str) -> Result<String> {
+            let mut ds = ds.clone();
+            log::warn!(
+                "Detecting out-dated fragment metadata, migrating dataset. \
+                        To disable migration, set LANCE_AUTO_MIGRATION=false"
```

Comment:
It's probably fine but this warning is a little too late :laughing: .  I guess users can always restore.

Reply:
Well this is for if they hit it in a production system. Hopefully they catch it on the first table it hits.